torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -333,11 +333,34 @@ class ProcessGroupNCCL : public ProcessGroup {
       return !value_.isNone();
     }
 
+    std::shared_ptr<at::cuda::CUDAStream> getfutureNCCLCallbackStream() {
+      return futureNCCLCallbackStream_;
+    }
+
+    bool hasRecordStreamCallback() const {
+        return record_stream_cb_ != nullptr;
+    }
+
+    void setRecordStreamCallback(
+        std::function<void(const at::IValue&, at::cuda::CUDAStream)>
+            record_stream_cb) {
+      record_stream_cb_ = std::move(record_stream_cb);
+    }
+
+    void recordStream() {
+      TORCH_CHECK(record_stream_cb_);
+      // Do not free the underlying data storage of value_ before its usage on
+      // futureNCCLCallbackStream_ finish.
+      record_stream_cb_(value_, *futureNCCLCallbackStream_);
+    }
+
    private:
     at::IValue value_;
     c10::DeviceIndex deviceIndex_;
     std::shared_ptr<std::vector<at::cuda::CUDAEvent>> cudaEvents_;
     std::shared_ptr<at::cuda::CUDAStream> futureNCCLCallbackStream_;
+    std::function<void(const at::IValue&, at::cuda::CUDAStream)>
```

Comment:
Nit: up to your preference, but you could do `using recordStreamCallback = std::function<void (....` to avoid having to write this twice. Although since it's used only twice not sure if it's worth it.

Reply:
I also had the same thought. After realizing that `std::function<at::IValue(void)> cb` has also appeared twice in this file, I decided not to create a separate alias.